src/app/services/statistics.service.ts

```diff
@@ -0,0 +1,79 @@
+import { Observable } from 'rxjs';
+import { CHART_DATA } from '../test-data/chart.data';
+import { QuestionModel } from './model/question.model';
+import { Injectable } from '@angular/core';
+import { ChartModel } from './model/chart.modal';
+import { QuizResultModel } from './model/quiz-result.model';
+import { decodeText } from '../utils/decode-html';
+import { QUESTIONS_SIZE, SECONDS, STORAGE_KEYS } from '../utils/constants';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class StatisticService {
+  get getQuizResult(): QuizResultModel | null {
+    return this.getFromStorage<QuizResultModel | null>(
+      STORAGE_KEYS.QUIZ_RESULT,
+      null
+    );
+  }
+
+  get getStatisticData(): ChartModel[] {
+    return this.getFromStorage<ChartModel[]>(
+      STORAGE_KEYS.CHART,
+      CHART_DATA
+    );
+  }
+
+  initDefaultStatistic() {
+    if (!localStorage.getItem(STORAGE_KEYS.CHART)) {
+      this.saveToStorage(STORAGE_KEYS.CHART, CHART_DATA);
+    }
+  }
+
+  initScore(questions$: Observable<QuestionModel[]>, answers: string[]): number {
+    let result: number = 0;
+  
+    questions$.subscribe((questions) => {
+      questions.forEach((question, index) => {
+        const isCorrect = decodeText(question.correct_answer) === answers[index];
+        if (isCorrect) {
+          result++;
+        }
+      });
+    }).unsubscribe();
+    return result;
+  }  
+
+  initResult(data: QuizResultModel): void {
+    this.saveToStorage(STORAGE_KEYS.QUIZ_RESULT, data);
+
+    const statistics = this.getStatisticData;
+    statistics[0].count++;
```

Comment:
What are statistics[0] and statistics[1] here? Same goes for additionalData

Reply:
it means that I push updated data to objects of statistics
statistics[0] - its how much quzzes I passes
statistics[1] - questions and additionalData[0] - right answers, additionalData[1] - wrong
statistics[2] - time

I know this is not such a good type of solution, but now I have no other ideas
Maybe you have some thoughts about how to do it better ?

Reply:
Instead of array let's create an object interface
{
passedQuizes: number, (or other type)
questions: {
    count: number,
    correct: number
},
time: number
}